application/LicenseAppService.go

```diff
@@ -145,25 +145,29 @@ func (s *LicenseAppService) ModifySeats(req ModifySeatAssignmentRequest) error {
 
 // ProcessOrgEntitledEvent handles the OrgEntitledEvent by storing the license and importing users
 func (s *LicenseAppService) ProcessOrgEntitledEvent(evt OrgEntitledEvent) error {
-	//lookupresourceson license & a lookusubjects on members of org and if both exist, return and log "import skipped"
-	licensed, err := s.seatRepo.IsLicensed(evt.OrgID)
+	// first, check if there's already a license/org to skip userImport after applying the new license, if necessary.
```

Comment:
If the logic only applies to whether or not users should be imported, then why not move `orgIsImported, err := s.seatRepo.IsImported(evt.OrgID)` below ApplyLicense (and optionally add a comment above ApplyLicense saying something like "ApplyLicense is not idempotent and will error if license exists for service and org.")

After all if `err = s.seatRepo.ApplyLicense(...` returns an error, there would never be a need to call `s.seatRepo.IsImported`.

Reply:
Logic here is:
1) check if _any_ license or user_ is there in the schema for a given org.
2) apply the new license.
3) then evaluate the result from 1) to see if the userImport is triggered or not. 

If we would put the IsImported check below the ApplyLicense call, it would **always** be the case that a license is there for the org, so users would **never** get imported.

If I understand you correctly, you are mixing up things here: This has nothing to do with the check if an existing license is tried to be written twice. For that, we have this precondition https://github.com/RedHatInsights/authz/pull/163/files#diff-4ebad1e00f0348022f44dd75c5fd3f9ff5ad4d620cbfc9592350ae23a699a2d1L414-L415 which results in an error on applyLicense, so the import would not be started as we return that error before (see line 162). Accomodating test: https://github.com/RedHatInsights/authz/pull/163/files#diff-22d71e93a01048789e68655f6f2131bf081a388f75de3f373d577682b9d51b58R163